specviz/core/annotation.py

```diff
@@ -6,29 +6,32 @@
 from qtpy.QtCore import QPointF
 from qtpy.QtGui import QPolygonF, QPen, QColor
 
+__all__ = ['LineIDMarkerProxy', 'LineIDMarker']
+
 orientations = {
     'horizontal': {'anchor': (0.5, 1), 'angle': 0},
     'vertical': {'anchor': (0, 0.5), 'angle': 90}
 }
 
 
 class LineIDMarkerProxy(object):
-    ''' A proxy class that is used in lieu of a real, full-blown Line1DMarker.
-
-        The LineIDMarker constructor calls pyqtgraph's TextItem constructor. Profiling
-        analysis showed that it's an expensive operation to perform. However, it has
-        to be called many times during the course of a zoom operation. This proxy,
-        by avoiding to call the Line1DMarker base class' constructor, speeds up
-        the entire process.
-
-        The idea is that this proxy is used to perform all the de-cluttering and
-        explicit clipping operations that take place before an actual marker can
-        be added to the plot. By postponing the instantiation of full-blown Line1DMarker
-        objects for the very end, only the ones actually showing up on screen are
-        instantiated. This saves an enormous amount of time not only on the constructor
-        phase, but also on the addItem and removeItem calls, which seem to generate in
-        turn an inordinate amount of calls to connect() and disconnect().
-    '''
+    """
+    A proxy class that is used in lieu of a real, full-blown Line1DMarker.
+
+    The LineIDMarker constructor calls pyqtgraph's TextItem constructor. Profiling
+    analysis showed that it's an expensive operation to perform. However, it has
+    to be called many times during the course of a zoom operation. This proxy,
+    by avoiding to call the Line1DMarker base class' constructor, speeds up
+    the entire process.
+
+    The idea is that this proxy is used to perform all the de-cluttering and
+    explicit clipping operations that take place before an actual marker can
+    be added to the plot. By postponing the instantiation of full-blown Line1DMarker
+    objects for the very end, only the ones actually showing up on screen are
+    instantiated. This saves an enormous amount of time not only on the constructor
+    phase, but also on the addItem and removeItem calls, which seem to generate in
+    turn an inordinate amount of calls to connect() and disconnect().
+    """
```

Comment:
Missing parameter documentation.

Reply:
This is used by the line labels stuff and will be replaced with Ivo's PRs.